packages/zoe/src/zoeService/startInstance.js

```diff
@@ -61,6 +63,28 @@ export const makeStartInstance = (
 
     const instance = makeInstanceHandle();
 
+    // Invitations whose descriptions contain any of the strings will be blocked
+    /** @type {Readonly<string[]>} */
+    let offerFilterStrings = [];
+    // if any string in the list matches, don't process the invitation
+    const isBlocked = string =>
+      offerFilterStrings.some(s => string.indexOf(s) > -1);
+
+    const setOfferFilter = strings => {
+      assert(Array.isArray(strings), X`${q(strings)} must be an Array`);
+      const proposedStrings = Array.from(strings);
+      assert(
+        proposedStrings.every(
+          s => typeof s === 'string' && isAlphanumeric.test(s),
+        ),
+        X`Blocked strings (${q(
+          strings,
+        )}) must be a (possibly empty) Array of alpha-numeric strings.`,
+      );
+
+      offerFilterStrings = proposedStrings;
```

Comment:
The original is ok. I'll explain why I suggest each of the changes below, but take only what you agree is an improvement.

```suggestion
    let offerFilterStrings = harden([]);
    // if any string in the list matches, don't process the invitation
    const isBlocked = string =>
      offerFilterStrings.some(s => string.indexOf(s) > -1);

    const setOfferFilter = strings => {
      harden(strings);
      fit(strings, M.arrayOf(M.string()));
      assert(
        strings.every(s => isAlphanumeric.test(s)),
        X`Blocked strings (${q(
          strings,
        )}) must be a (possibly empty) Array of alpha-numeric strings.`,
      );

      offerFilterStrings = strings;
```

I see a declaration of `ReadOnly` and code consistent with the array actually being immutable. The original copies the incoming array into a new array which it then never mutates. So it seems stronger than ReadOnly anyway.

But TS is unsound. If you want to ensure that the array isn't mutated, you can harden it. Fortunately, a hardened array of strings is a CopyArray, and so a Passable, and so something we can pattern match on. The pattern match above asserts in one line what took several lines to say. Unfortunately, we cannot express the alphanumeric constraint in the pattern language so that still needs to be explicit.

A possible reason to prefer your code over my suggested pattern match: Better diagnostics. My pattern matching logic doesn't understand the purpose of the match, and so cannot give diagnostics that are as informative.

Once we know that `strings` is a CopyArray of strings, we know it is safe enough that we don't need to copy it into `proposedStrings`. OTOH, this hardens whatever `strings` argument was passed in by our caller, which we might want to avoid. In that case, we can start with
```js
const proposedStrings = harden([...strings]);
```
and then do everything else with `proposedStrings` more like your original. Note that `[...foo]` is essentially the same as `Array.from(foo)` but IMO reads nicer.

Reply:
I did not change the code for `isBlocked` above because I discuss the substring match criteria separately.

Reply:
Current state looks good. Resolving.